api/proto/teleport/machineid/v1/bot_service.proto

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.machineid.v1;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/field_mask.proto";
+import "teleport/machineid/v1/bot.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1;machineidv1";
+
+// BotService provides methods to manage Teleport Bots
+service BotService {
+  // GetBot is used to query a Bot resource by its name.
+  //
+  // This will return a NotFound error if the specified Bot does not exist.
+  rpc GetBot(GetBotRequest) returns (GetBotResponse);
+  // ListBots is used to query Bots.
+  //
+  // Follows the pagination semantics of
+  // https://cloud.google.com/apis/design/standard_methods#list.
+  rpc ListBots(ListBotsRequest) returns (ListBotsResponse);
+  // CreateBot is used to create a Bot.
+  //
+  // This will return an error if a Bot by that name already exists.
+  rpc CreateBot(CreateBotRequest) returns (CreateBotResponse);
+  // UpdateBot is used to modify an existing Bot.
+  rpc UpdateBot(UpdateBotRequest) returns (UpdateBotResponse);
+  // UpsertBot is used to create or replace an existing Bot.
+  //
+  // Prefer using CreateBot and UpsertBot.
+  rpc UpsertBot(UpsertBotRequest) returns (UpsertBotResponse);
+  // DeleteBot is used to delete a specific Bot.
+  //
+  // This will return a NotFound error if the specified Bot does not exist.
+  rpc DeleteBot(DeleteBotRequest) returns (google.protobuf.Empty);
+}
+
+// The request for CreateBot.
+message CreateBotRequest {
+  // The bot to create.
+  Bot bot = 1;
+}
+
+// The response for CreateBot.
+message CreateBotResponse {
```

Comment:
There's some debate over on the RFD about whether these "wrapper" messages should be used or if the resource should just be the RPCs direct return. We may wish to wait for a decision to be made before we merge this PR.

Reply:
Great - I'll drop the wrappers from here.

Reply:
[fcff785](https://github.com/gravitational/teleport/pull/35216/commits/fcff78555031f4365dd9bfbe053a91c6bfea1d91)